### ISSUE TITLE
idx_i is not differentiable

### ISSUE THREAD

**Issue by pcshih** [atulkum/co-attention#1]
https://github.com/atulkum/co-attention/blob/5dd994b74a591804569077e03906a4360ed06ffb/code/model.py#L214

As shown above, idx_i is not differentiable.
How can the gradient backprop. through?

Thank you.

**Comment by atulkum** [comment]
You might find this discussion helpful https://discuss.pytorch.org/t/confused-about-torch-max-and-gradient/14283
